src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.Android.cs

```diff
@@ -0,0 +1,391 @@
+                }
+
+                ReadIndex(tzFilePath, fs, header.indexOffset, header.dataOffset, buffer);
+            }
+
+            [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2087:UnrecognizedReflectionPattern",
+                Justification = "Implementation detail of Android TimeZone")]
+            private unsafe T ReadAt<T>(string tzFilePath, Stream fs, long position, Span<byte> buffer)
+                where T : struct
+            {
+                int size = Marshal.SizeOf(typeof(T));
+                Debug.Assert(buffer.Length >= size);
+
+                fs.Position = position;
+                int numBytesRead;
+                if ((numBytesRead = fs.Read(buffer)) < size)
+                {
+                    throw new InvalidOperationException(SR.Format(SR.InvalidOperation_ReadTZError, tzFilePath, position, size, numBytesRead, size));
+                }
+
+                fixed (byte* b = buffer)
+                {
+                    return (T)Marshal.PtrToStructure((IntPtr)b, typeof(T))!; // Is ! the right way to handle Unboxing a possibly null value. Should there be some check instead?
+                }
+            }
+
+            private static int NetworkToHostOrder(int value)
+            {
+                if (!BitConverter.IsLittleEndian)
+                    return value;
+
+                return (((value >> 24) & 0xFF) | ((value >> 08) & 0xFF00) | ((value << 08) & 0xFF0000) | ((value << 24)));
+            }
+
+            private static unsafe int GetStringLength(sbyte* s, int maxLength)
+            {
+                int len;
+                for (len = 0; len < maxLength; len++, s++)
+                {
+                    if (*s == 0)
+                    {
+                        break;
+                    }
+                }
+                return len;
+            }
+
+            [MemberNotNull(nameof(_ids))]
+            [MemberNotNull(nameof(_byteOffsets))]
+            [MemberNotNull(nameof(_lengths))]
+            private unsafe void ReadIndex(string tzFilePath, Stream fs, int indexOffset, int dataOffset, Span<byte> buffer)
+            {
+                int indexSize = dataOffset - indexOffset;
+                int entrySize = Marshal.SizeOf(typeof(AndroidTzDataEntry));
+                int entryCount = indexSize / entrySize;
+
+                _byteOffsets = new int[entryCount];
+                _ids = new string[entryCount];
+                _lengths = new int[entryCount];
+
+                for (int i = 0; i < entryCount; ++i)
+                {
+                    AndroidTzDataEntry entry = ReadAt<AndroidTzDataEntry>(tzFilePath, fs, indexOffset + (entrySize*i), buffer);
+                    var p = (sbyte*)entry.id;
+
+                    _byteOffsets![i] = NetworkToHostOrder(entry.byteOffset) + dataOffset;
+                    _ids![i] = new string(p, 0, GetStringLength(p, 40), Encoding.ASCII);
```

Comment:
It seems like `Encoding.ASCII` is the intended encoding. `new string(p)` would use `Default`.

Reply:
I don't see any encoding logic in https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/String.cs#L191-L206

Reply:
Right, my concern was more whether or not the tzdata bytes was intended to be parsed with ASCII encoding since the mono/mono implementation did that, but I'm not sure why. I'll change to `new string(p)`